authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -0,0 +1,1006 @@
+    return rangerClient.createGroup(VXGroup.builder().withName(group.name()).build());
+  }
+
+  @Override
+  public Boolean onGroupRemoved(Group group) throws RuntimeException {
+    VXGroupList list = rangerClient.searchGroup(ImmutableMap.of("name", group.name()));
+    if (list.getListSize() == 0) {
+      LOG.warn("The group({}) is not exists in the Ranger!", group);
+      return Boolean.FALSE;
+    }
+    return rangerClient.deleteGroup(list.getList().get(0).getId());
+  }
+
+  @Override
+  public Boolean onGroupAcquired(Group group) {
+    VXGroupList vxGroupList = rangerClient.searchGroup(ImmutableMap.of("name", group.name()));
+    if (vxGroupList.getListSize() == 0) {
+      LOG.warn("The group({}) is not exists in the Ranger!", group);
+      return Boolean.FALSE;
+    }
+    return Boolean.TRUE;
+  }
+
+  private boolean checkRangerRole(String roleName) throws AuthorizationPluginException {
+    try {
+      rangerClient.getRole(roleName, RANGER_ADMIN_NAME, rangerServiceName);
+    } catch (RangerServiceException e) {
+      throw new AuthorizationPluginException(e);
+    }
+    return true;
+  }
+
+  private GrantRevokeRoleRequest createGrantRevokeRoleRequest(
+      String roleName, String userName, String groupName) {
+    Set<String> users;
+    if (userName == null || userName.isEmpty()) {
+      users = new HashSet<>();
+    } else {
+      users = new HashSet<>(Arrays.asList(userName));
+    }
+    Set<String> groups;
+    if (groupName == null || groupName.isEmpty()) {
+      groups = new HashSet<>();
+    } else {
+      groups = new HashSet<>(Arrays.asList(groupName));
+    }
+
+    GrantRevokeRoleRequest roleRequest = new GrantRevokeRoleRequest();
+    roleRequest.setUsers(users);
+    roleRequest.setGroups(groups);
+    roleRequest.setGrantor(RANGER_ADMIN_NAME);
+    roleRequest.setTargetRoles(new HashSet<>(Arrays.asList(roleName)));
+    return roleRequest;
+  }
+
+  /** Create a Ranger role if the role does not exist. */
+  private RangerRole createRangerRoleIfNotExists(String roleName) {
+    RangerRole rangerRole = null;
+    try {
+      rangerRole = rangerClient.getRole(roleName, RANGER_ADMIN_NAME, rangerServiceName);
+    } catch (RangerServiceException e) {
+      // ignore exception, If the role does not exist, then create it.
+    }
+    try {
+      if (rangerRole == null) {
+        rangerRole = new RangerRole(roleName, MANAGED_BY_GRAVITINO, null, null, null);
+        rangerClient.createRole(rangerServiceName, rangerRole);
+      }
+    } catch (RangerServiceException e) {
+      throw new RuntimeException(e);
+    }
+    return rangerRole;
+  }
+
+  /**
+   * Find the managed policy for the metadata object.
+   *
+   * @param metadataObject The metadata object to find the managed policy.
+   * @return The managed policy for the metadata object.
+   */
+  @VisibleForTesting
+  public RangerPolicy findManagedPolicy(MetadataObject metadataObject)
```

Comment:
Could you handle the case?
`catalog READ_TABLE`
`metalake  READ_TABLE`

Reply:
We can consider this case in the `0.6.1`.